diracx-routers/src/diracx/routers/__init__.py

```diff
@@ -442,3 +448,53 @@ async def db_transaction(db: T2) -> AsyncGenerator[T2]:
         if reason := await is_db_unavailable(db):
             raise DBUnavailable(reason)
         yield db
+
+
+class ClientMinVersionCheckMiddleware(BaseHTTPMiddleware):
+    """Custom FastAPI middleware to verify that
+    the client has the required minimum version.
+    """
+
+    def __init__(self, app: FastAPI):
+        super().__init__(app)
+        self.min_client_version = get_min_client_version()
+        self.parsed_min_client_version = parse(self.min_client_version)
+
+    async def dispatch(self, request: Request, call_next) -> Response:
+        client_version = request.headers.get("DiracX-Client-Version")
+        if client_version and self.is_version_too_old(client_version):
+            # When comes from Swagger or Web, there is no client version header.
+            # This is not managed here.
+            raise HTTPException(
+                status_code=HTTPStatus.UPGRADE_REQUIRED,
+                detail=f"Client version ({client_version}) not recent enough (>= {self.min_client_version}). Upgrade.",
+            )
+
+        response = await call_next(request)
+        return response
+
+    def is_version_too_old(self, client_version: str) -> bool | None:
+        """Verify that client version is ge than min."""
+        try:
+            return parse(client_version) < self.parsed_min_client_version
+        except InvalidVersion as iv_exc:
+            raise HTTPException(
+                status_code=HTTPStatus.BAD_REQUEST,
+                detail=f"Invalid version string: '{client_version}'",
+            ) from iv_exc
+
+
+def get_min_client_version():
+    """Extracting min client version from entry_points and searching for extension."""
+    matched_entry_points: EntryPoints = entry_points(group="diracx.min_client_version")
```

Comment:
This just made me realize that it is only compatible with versioning the vanilla diracx client, and not the extension.  You already spent a substantial amount of time on this PR, so I will not ask you to make it extension-compatible :laughing:  I'll just open an issue with it